DataMigration/src/main/java/org/sagebionetoworks/dian/datamigration/BridgeUtil.java

```diff
@@ -0,0 +1,570 @@
+import java.util.Map;
+
+public class BridgeUtil {
+
+    // Do not change the name of these two variables, as it is used in CI to replace the values
+    // However, to get the app to run, you must provide your own email/password,
+    // NEVER commit your credentials to github
+    public static String bridgeAdminEmail = "";
+    public static String bridgeAdminPassword = "";
+
+    // TODO: mdephillips 9/19/21 these will need switched to the official study created on bridge
+    private static String STUDY_ID = "dian-validation-study";
+    private static String BRIDGE_ID = "dian-validation";
+    // TODO: mdephillips 9/19/21 this will need switched to an empty array list for production
+    private static List<String> userDataGroups = Collections.singletonList("test_user");
+
+    public static String MIGRATED_REPORT_ID = "Migrated";
+    public static String AVAILABILITY_REPORT_ID = "Availability";
+    public static String TEST_SCHEDULE_REPORT_ID = "TestSchedule";
+    public static String COMPLETED_TESTS_REPORT_ID = "CompletedTests";
+
+    private static String base = "https://webservices.sagebridge.org";
+    private static String signIn = "/v3/auth/signIn";
+    private static String participants = "/v3/participants";
+
+    private static ZonedDateTime singletonReportDate =
+            ZonedDateTime.ofInstant(Instant.EPOCH, ZoneOffset.UTC);
+
+    private static String bridgeUrl(String suffix) {
+        return base + suffix;
+    }
+
+    private static String userUrl(String userId) {
+        return base + "/v3/participants/" + userId;
+    }
+
+    private static String getReportUrl(String userId, String reportId) {
+        return base + "/v3/participants/" + userId + "/reports/" + reportId;
+    }
+
+    private static String postReportUrl(String userId, String reportId) {
+        return base + "/v4/participants/" + userId + "/reports/" + reportId;
+    }
+
+    private static String getAllUserUrl(int offsetBy, int pageSize) {
+        return bridgeUrl("/v3/participants") + "?offsetBy=" +
+                offsetBy + "&pageSize=" + pageSize;
+    }
+
+    /**
+     * @return authentication token
+     * @throws MalformedURLException if sign in URL is bad
+     * @throws IOException if there is an HTTP error
+     */
+    public static String authenticate() throws MalformedURLException, IOException {
+        EmailSignIn emailSignIn = new EmailSignIn();
+        emailSignIn.email = bridgeAdminEmail;
+        emailSignIn.password = bridgeAdminPassword;
+        emailSignIn.appId = BRIDGE_ID;
+
+        ObjectWriter ow = new ObjectMapper().writer();
+        String json = ow.writeValueAsString(emailSignIn);
+
+        String response = bridgePOST(bridgeUrl(signIn), null, json);
+
+        ObjectMapper mapper = new ObjectMapper();
+        SignInResponse responseObj = mapper.readValue(response, SignInResponse.class);
+
+        return responseObj.sessionToken;
+    }
+
+    public static <T> T getSingletonReport(
```

Comment:
I didn't know about this API, is it new?  

The Sage mobile team has been using this paradigm for years to store "singleton" reports which is where I got the idea.  Hopefully this will be used in the new mobile sdk.